{
  "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/16936",
  "repository_url": "https://api.github.com/repos/ethereum/go-ethereum",
  "labels_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/16936/labels{/name}",
  "comments_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/16936/comments",
  "events_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/16936/events",
  "html_url": "https://github.com/ethereum/go-ethereum/issues/16936",
  "id": 330930763,
  "node_id": "MDU6SXNzdWUzMzA5MzA3NjM=",
  "number": 16936,
  "title": "Ethers are wiped out in account during sendTransaction or migrations",
  "user": {
    "login": "samaujs",
    "id": 13311002,
    "node_id": "MDQ6VXNlcjEzMzExMDAy",
    "avatar_url": "https://avatars.githubusercontent.com/u/13311002?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/samaujs",
    "html_url": "https://github.com/samaujs",
    "followers_url": "https://api.github.com/users/samaujs/followers",
    "following_url": "https://api.github.com/users/samaujs/following{/other_user}",
    "gists_url": "https://api.github.com/users/samaujs/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/samaujs/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/samaujs/subscriptions",
    "organizations_url": "https://api.github.com/users/samaujs/orgs",
    "repos_url": "https://api.github.com/users/samaujs/repos",
    "events_url": "https://api.github.com/users/samaujs/events{/privacy}",
    "received_events_url": "https://api.github.com/users/samaujs/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [
    {
      "id": 856638432,
      "node_id": "MDU6TGFiZWw4NTY2Mzg0MzI=",
      "url": "https://api.github.com/repos/ethereum/go-ethereum/labels/status:inactive",
      "name": "status:inactive",
      "color": "ffffff",
      "default": false,
      "description": null
    }
  ],
  "state": "closed",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 3,
  "created_at": "2018-06-10T01:01:10Z",
  "updated_at": "2019-07-24T14:25:18Z",
  "closed_at": "2019-07-24T14:25:18Z",
  "author_association": "NONE",
  "active_lock_reason": null,
  "body": "Hi there,\r\n\r\nplease note that this is an issue tracker reserved for bug reports and feature requests.\r\n\r\nFor general questions please use the gitter channel or the Ethereum stack exchange at https://ethereum.stackexchange.com.\r\n\r\n#### System information\r\nGeth version:\r\nVersion: 1.8.10-stable\r\nGit Commit: eae63c511ceafab14b92e274c1b18bf1700e2d3d\r\nArchitecture: amd64\r\nProtocol Versions: [63 62]\r\nNetwork Id: 1\r\nGo Version: go1.10\r\n\r\nCommit hash : (if `develop`)\r\n\r\nOperating System:\r\nUbuntu 16.04.4 LTS (GNU/Linux 4.4.0-1060-aws x86_64)\r\n\r\nEthereum client:\r\nTruffle version (truffle version): Truffle v4.1.11 (core: 4.1.11), Solidity v0.4.24 (solc-js)\r\nnode version (node --version): v9.11.1\r\nnpm version (npm --version): 5.6.0\r\n\r\n#### Expected behaviour\r\nThe wipe-out of all ethers has not been encountered before.\r\n\r\n#### Actual behaviour\r\nAll ethers are wipeout and these mysterious transactions (for account address, 0x25376e3269172e1811DBF053460dC5b090499f89) are depicted in blocks 2429765, 2426633, 2426436 and 2426257.\r\n\r\n#### Steps to reproduce the behaviour\r\nLost all rinkeby ethers during truffle compile and/or sendTransaction in geth console.\r\n(A) Rinkeby coinbase is 0x25376e3269172e1811DBF053460dC5b090499f89\r\n(B) eth.sendTransaction({from: eth.accounts[0], to: eth.accounts[1], value: web3.toWei(5, \"ether\")})\r\n\r\nPlease provide the shortest amount of steps to reproduce your issue.\r\n(1) geth --rinkeby --rpc --rpcport \"8545\" --rpcaddr \"0.0.0.0\" --rpccorsdomain \"http://localhost:8000\" --rpcapi=\"personal,eth,network,web3,net\" --ipcpath \"~/.ethereum/geth.ipc\"\r\n(2) truffle migrate --compile-all --reset --network rinkeby\r\n(3) geth attach\r\n\r\nIn the Geth console\r\n(1) eth.sendTransaction({from: eth.accounts[0], to: eth.accounts[1], value: web3.toWei(5, \"ether\")})\r\n#### Backtrace\r\nThe following details for your reference:\r\n\r\n(1) The account address are generated with the command \"geth --rinkeby account new\";\r\n\r\n(2) I did not publish my private key; other than using the address to obtain rinkeby fauet as per instruction from google plus;\r\n\r\n(3) I have been using these addresses in the last few months (the first 3 addresses are created months ago and the other 2 are newly created for testing since I lost all the ethers mysteriously) :\r\n\r\n[\"0x25376e3269172e1811dbf053460dc5b090499f89\", \"0x96849a6a47ef7aa4a4f5eca9ff8d3242ffbe271c\", \"0x2efc9b04d1595e246f482df8c86a4ae2fe8e0792\", \"0xb856b88ed8b73ef697dc92b1a92e7e89d8ff89fe\", \"0x00291f19e140bf4a669ba8252371d404075fa92d\"]\r\n\r\n(4) This lost of ethers started recently from block 2426257<https://rinkeby.etherscan.io/block/2426257> of the address \"0x25376e3269172e1811DBF053460dC5b090499f89\".  This is the first encounter.\r\n(5)  One typical series of action is when I obtain the faucet in block 2429754, compile and migrate in block 2429764 and ethers being mysteriously transferred out in block 242965.  This is what happens about 10 hours ago.\r\nTxHash  Block   Age     From            To      Value   [TxFee]\r\n0x56f063ed3634007d3d93f8d9733f2225be9c3a3cb706a0809a7ee93633912610<https://rinkeby.etherscan.io/tx/0x56f063ed3634007d3d93f8d9733f2225be9c3a3cb706a0809a7ee93633912610>  2429765<https://rinkeby.etherscan.io/block/2429765>     10 hrs 37 mins ago      0x25376e3269172e1811dbf053460dc5b090499f89      OUT     0x7097f41f1c1847d52407c629d0e0ae0fdd24fd58<https://rinkeby.etherscan.io/address/0x7097f41f1c1847d52407c629d0e0ae0fdd24fd58>     2.883086186000018 Ether 0.08916761\r\n0xa8c4f77f6a20d87c055bdf556d622554c4bdd0cb5a39680cff819f5ae36433ba<https://rinkeby.etherscan.io/tx/0xa8c4f77f6a20d87c055bdf556d622554c4bdd0cb5a39680cff819f5ae36433ba>  2429764<https://rinkeby.etherscan.io/block/2429764>     10 hrs 37 mins ago      0x25376e3269172e1811dbf053460dc5b090499f89      OUT     [https://rinkeby.etherscan.io/images/id/application-table.png]  Contract Creation<https://rinkeby.etherscan.io/address/0x9374a4b6bb9834cc4560313153269222a7bc291e>      0 Ether 0.0277462\r\n0x60b3158a165eb2ce923ff930e0bc02848a203c563ba40addd3c8001f647d7aca<https://rinkeby.etherscan.io/tx/0x60b3158a165eb2ce923ff930e0bc02848a203c563ba40addd3c8001f647d7aca>  2429754<https://rinkeby.etherscan.io/block/2429754>     10 hrs 40 mins ago      0x31b98d14007bdee637298086988a0bbd31184523<https://rinkeby.etherscan.io/address/0x31b98d14007bdee637298086988a0bbd31184523>       IN    0x25376e3269172e1811dbf053460dc5b090499f89      3 Ether 0.000021\r\n(6) The above actions, I have tried multiple times.  It happens even when I transferred ether between my accounts in geth console with the command \"eth.sendTransaction({from: eth.accounts[4], to: eth.accounts[3], value: web3.toWei(1, \"ether\")})\".\r\n\r\n(7) Lastly, I have tried again about an hour ago, I cannot get the exact time from etherscan (as it is currently out of sync).  Once I obtain the faucet, I made a transfer of 3.9 eth (from \"0xb856b88ed8b73ef697dc92b1a92e7e89d8ff89fe\" to \"0x25376e3269172e1811DBF053460dC5b090499f89\"), compile and migrate that works fine as usual before the problem.  But, within 15-20 minutes, the remaining ethers are transferred away again leaving the coinbase (0x25376e3269172e1811DBF053460dC5b090499f89) with 0 ether remaining.\r\n\r\n<https://rinkeby.etherscan.io/block/2426257>\r\nâ€¦\r\n\r\n--\r\n\r\n  Hope the above details are sufficient for locating the problem.\r\n\r\n  Please advise what should I do next?  Should I change to a new coinbase and don't use the above addresses anymore?  Also, opened an issue in truffle (Ethers are wiped out in account during sendTransaction or migrations #997) and advise to do likewise here.  Thanks. ",
  "closed_by": {
    "login": "stale[bot]",
    "id": 26384082,
    "node_id": "MDM6Qm90MjYzODQwODI=",
    "avatar_url": "https://avatars.githubusercontent.com/in/1724?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/stale%5Bbot%5D",
    "html_url": "https://github.com/apps/stale",
    "followers_url": "https://api.github.com/users/stale%5Bbot%5D/followers",
    "following_url": "https://api.github.com/users/stale%5Bbot%5D/following{/other_user}",
    "gists_url": "https://api.github.com/users/stale%5Bbot%5D/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/stale%5Bbot%5D/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/stale%5Bbot%5D/subscriptions",
    "organizations_url": "https://api.github.com/users/stale%5Bbot%5D/orgs",
    "repos_url": "https://api.github.com/users/stale%5Bbot%5D/repos",
    "events_url": "https://api.github.com/users/stale%5Bbot%5D/events{/privacy}",
    "received_events_url": "https://api.github.com/users/stale%5Bbot%5D/received_events",
    "type": "Bot",
    "site_admin": false
  },
  "reactions": {
    "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/16936/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/16936/timeline",
  "performed_via_github_app": null,
  "state_reason": "completed"
}
[
  {
    "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/comments/396179383",
    "html_url": "https://github.com/ethereum/go-ethereum/issues/16936#issuecomment-396179383",
    "issue_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/16936",
    "id": 396179383,
    "node_id": "MDEyOklzc3VlQ29tbWVudDM5NjE3OTM4Mw==",
    "user": {
      "login": "karalabe",
      "id": 129561,
      "node_id": "MDQ6VXNlcjEyOTU2MQ==",
      "avatar_url": "https://avatars.githubusercontent.com/u/129561?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/karalabe",
      "html_url": "https://github.com/karalabe",
      "followers_url": "https://api.github.com/users/karalabe/followers",
      "following_url": "https://api.github.com/users/karalabe/following{/other_user}",
      "gists_url": "https://api.github.com/users/karalabe/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/karalabe/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/karalabe/subscriptions",
      "organizations_url": "https://api.github.com/users/karalabe/orgs",
      "repos_url": "https://api.github.com/users/karalabe/repos",
      "events_url": "https://api.github.com/users/karalabe/events{/privacy}",
      "received_events_url": "https://api.github.com/users/karalabe/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2018-06-11T09:23:23Z",
    "updated_at": "2018-06-11T09:23:23Z",
    "author_association": "MEMBER",
    "body": "Does Truffle unlock you account when doing migrations? Did you run your node with the RPC port open publicly? If yes, anyone from the internet could have been connected to your node, and signed a number of transactions with your account to both sweep its current funds as well as to transfer any future funds received. There are active attacks on mainnet and all testnets that exploit vulnerable setups.\r\n\r\nPlease make sure that you don't ever run publicly reachable RPC, and ideally don't unlock your account, rather use the one-shot personal.sendTransaction.",
    "reactions": {
      "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/comments/396179383/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/comments/396252251",
    "html_url": "https://github.com/ethereum/go-ethereum/issues/16936#issuecomment-396252251",
    "issue_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/16936",
    "id": 396252251,
    "node_id": "MDEyOklzc3VlQ29tbWVudDM5NjI1MjI1MQ==",
    "user": {
      "login": "samaujs",
      "id": 13311002,
      "node_id": "MDQ6VXNlcjEzMzExMDAy",
      "avatar_url": "https://avatars.githubusercontent.com/u/13311002?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/samaujs",
      "html_url": "https://github.com/samaujs",
      "followers_url": "https://api.github.com/users/samaujs/followers",
      "following_url": "https://api.github.com/users/samaujs/following{/other_user}",
      "gists_url": "https://api.github.com/users/samaujs/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/samaujs/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/samaujs/subscriptions",
      "organizations_url": "https://api.github.com/users/samaujs/orgs",
      "repos_url": "https://api.github.com/users/samaujs/repos",
      "events_url": "https://api.github.com/users/samaujs/events{/privacy}",
      "received_events_url": "https://api.github.com/users/samaujs/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2018-06-11T13:57:05Z",
    "updated_at": "2018-06-11T13:57:05Z",
    "author_association": "NONE",
    "body": "No, I do not unlock my account with truffle. This happens during the time when I unlocked the affected accounts in geth. Based on the logs, typically happens when I am doing contract migrations or even just a test of sending eth within geth for a period of 300 seconds. The incidents happen right after a contract migration or send transaction is made.\r\nThe RPC port is opened with geth when I am trying to run the eth explorer and statistics. Is there a secure way to have the eth explorer and statistics running in a remote server whereby I can access publicly?\r\nYes, I am using the one-shot send transaction in a DApp.\r\nThanks for sharing.",
    "reactions": {
      "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/comments/396252251/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/comments/501291333",
    "html_url": "https://github.com/ethereum/go-ethereum/issues/16936#issuecomment-501291333",
    "issue_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/16936",
    "id": 501291333,
    "node_id": "MDEyOklzc3VlQ29tbWVudDUwMTI5MTMzMw==",
    "user": {
      "login": "stale[bot]",
      "id": 26384082,
      "node_id": "MDM6Qm90MjYzODQwODI=",
      "avatar_url": "https://avatars.githubusercontent.com/in/1724?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/stale%5Bbot%5D",
      "html_url": "https://github.com/apps/stale",
      "followers_url": "https://api.github.com/users/stale%5Bbot%5D/followers",
      "following_url": "https://api.github.com/users/stale%5Bbot%5D/following{/other_user}",
      "gists_url": "https://api.github.com/users/stale%5Bbot%5D/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/stale%5Bbot%5D/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/stale%5Bbot%5D/subscriptions",
      "organizations_url": "https://api.github.com/users/stale%5Bbot%5D/orgs",
      "repos_url": "https://api.github.com/users/stale%5Bbot%5D/repos",
      "events_url": "https://api.github.com/users/stale%5Bbot%5D/events{/privacy}",
      "received_events_url": "https://api.github.com/users/stale%5Bbot%5D/received_events",
      "type": "Bot",
      "site_admin": false
    },
    "created_at": "2019-06-12T14:10:40Z",
    "updated_at": "2019-06-12T14:10:40Z",
    "author_association": "NONE",
    "body": "This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.\n",
    "reactions": {
      "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/comments/501291333/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  }
]
